internal/server/v1/event/event.go

```diff
@@ -39,12 +63,38 @@ func Collection(
 	if !util.CheckValidDCIDs([]string{in.GetAffectedPlaceDcid()}) {
 		return nil, status.Errorf(codes.InvalidArgument, "Invalid DCID")
 	}
+	filterProp := in.GetFilterProp()
+	filterLimit := in.GetFilterLimit()
+	filterUnit := in.GetFilterUnit()
+
+	// Merge all the affected places in the final result.
+	affectedPlaces := []string{}
+	if in.GetAffectedPlaceDcid() == "Earth" {
```

Comment:
we should do this for continents here too.

Reply:
Done